### PR TITLE
[READY] MAINT - Rename ``scal`` by ``dnorm``

### LINKED ISSUE #247: MAINT - Rename ``scal`` by ``dnorm``
To Improve naming.

Following https://github.com/mathurinm/celer/pull/230#discussion_r860675015

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**
fix #247 

- [x] rename ``scal`` by ``dnorm``
- [x] rename ``tmp`` by ``theta_scaling``
- [x] rename ``norm_Xtheta`` and ``norm_Xtheta_acc``

### CODE REVIEW

File: celer/group_fast.pyx [hunk 1]
```diff
@@ -237,24 +237,24 @@ cpdef celer_grp(
...
 
-        scal = dnorm_grp(
+        dnorm = dnorm_grp(
             is_sparse, theta, grp_ptr, grp_indices, X, X_data, X_indices,
             X_indptr, X_mean, weights, n_groups, dummy_C, center)
 
-        if scal > alpha:
-            tmp = alpha / scal
+        if dnorm > alpha:
+            tmp = alpha / dnorm
```
Comment (mathurinm) [inline celer/group_fast.pyx:245]: can you also rename tmp to `theta_scaling` ? (name open to discussion :) )

Comment (Badr-MOUFAD) [reply celer/group_fast.pyx:245]: I agree with the name

**Comment by Badr-MOUFAD** [comment]
Can we fix the naming for the prox newton solver also? Indeed, we follow a similar logic yet we adopt a different naming convention.

https://github.com/mathurinm/celer/blob/4230db117a916e5158cfae85b2cd1a7249cf5475/celer/PN_logreg.pyx#L50

Also, we have two variables (``norm_Xtheta``, ``norm_Xtheta_acc``) that perform the same task. We better keep only one of them

File: celer/multitask_fast.pyx [hunk 2]
```diff
@@ -70,7 +70,6 @@ cdef void set_prios_mtl(
         int * n_screened) nogil:
     cdef int j, k
     cdef int inc = 1
-    cdef floating tmp
```
Comment (Badr-MOUFAD) [inline celer/multitask_fast.pyx:73]: Forgive these changes. (Might be considered out of the scoop of the PR)
They shall be fixed here or never!

File: celer/multitask_fast.pyx [hunk 3]
```diff
@@ -191,25 +190,23 @@ def celer_mtl(
         # theta = R :
         fcopy(&n_obs, &R[0, 0], &inc, &theta[0, 0], &inc)
 
-        tmp = 1.
```
Comment (Badr-MOUFAD) [inline celer/multitask_fast.pyx:194]: Same, forgive these changes.

**Comment by mathurinm** [comment]
+1 to harmonize with the same logic in prox newton solver

**Review by mathurinm — APPROVED** [review]

**Pull request merged**